pallets/auctions/src/types.rs

```diff
@@ -0,0 +1,243 @@
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+use sp_runtime::RuntimeDebug;
+use ternoa_primitives::marketplace::MarketplaceId;
+
+#[derive(Encode, Decode, Clone, PartialEq, RuntimeDebug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+/// Structure to store Auction data
+pub struct AuctionData<AccountId, BlockNumber, BalanceCaps>
+where
+    AccountId: Clone + Default,
+    BalanceCaps: Clone + Default,
+{
+    pub creator: AccountId,
+    pub start_block: BlockNumber,
+    pub end_block: BlockNumber,
+    pub start_price: BalanceCaps,
+    pub buy_it_price: Option<BalanceCaps>,
+    pub bidders: SortedBidderList<AccountId, BalanceCaps>,
+    pub marketplace_id: MarketplaceId,
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, RuntimeDebug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+/// wrapper type to store sorted list of all bids
+/// The wrapper exists to ensure a queue implementation of sorted bids
+pub struct SortedBidderList<AccountId, BalanceCaps>(pub Vec<(AccountId, BalanceCaps)>);
+
+impl<AccountId, BalanceCaps> SortedBidderList<AccountId, BalanceCaps>
+where
+    AccountId: std::cmp::Ord + Clone,
+    BalanceCaps: std::cmp::PartialOrd,
+{
+    pub const MAX_COUNT: usize = 10;
+
+    /// Create a new empty bidders list
+    pub fn new() -> Self {
+        Self(Vec::new())
+    }
+
+    /// Insert a new bid to the list
+    ///
+    /// The function ensures that the new bid is always higher than the current highest bid
+    /// If the insert action will cause an overflow, the lowest bid is removed and returned
+    /// If value lower than current highest bid is passed, the function will panic
+    pub fn insert_new_bid(
+        &mut self,
+        account_id: AccountId,
+        value: BalanceCaps,
+    ) -> Option<(AccountId, BalanceCaps)> {
+        // ensure the new bid is larger than current highest bid
+        if let Some(current_highest_bid) = self.get_current_highest_bid() {
```

Comment:
You can safely remove this check. The list should not care what kind of value we insert.

Reply:
I think it makes sense to keep this check since the struct is named `SortedBiddersList`, there should be a check to ensure its inserted as sorted or sorted on insertion. The check seemed easier :)